File: app/src/main/java/io/github/hyuwah/muslimcompanionapp/View/MainActivity.java
package io.github.hyuwah.muslimcompanionapp.View;

import android.content.Intent;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import android.os.Bundle;
import com.google.android.material.navigation.NavigationView;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.util.Log;
import android.view.MenuItem;
import android.widget.FrameLayout;
import butterknife.BindView;
import butterknife.ButterKnife;
import io.github.hyuwah.muslimcompanionapp.R;

public class MainActivity extends AppCompatActivity
    implements NavigationView.OnNavigationItemSelectedListener {

  @BindView(R.id.toolbar)
  Toolbar toolbar;
  @BindView(R.id.drawer_layout)
  DrawerLayout drawer;
  @BindView(R.id.nav_view)
  NavigationView navigationView;
  @BindView(R.id.view_fragment)
  FrameLayout viewFragment;

  FragmentManager fragmentManager;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    ButterKnife.bind(this);

    // Setup Toolbar & Drawer Layout
    setSupportActionBar(toolbar);
    ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
        this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
    drawer.addDrawerListener(toggle);
    toggle.syncState();
    navigationView.setNavigationItemSelectedListener(this);
    navigationView.setCheckedItem(R.id.nav_ayat_fetcher);

    fragmentManager = getSupportFragmentManager();

    // Set default screen
    fragmentManager.beginTransaction().replace(R.id.view_fragment, new AyahFetcherView()).commit();

  }


  @Override
  public void onBackPressed() {
    if (drawer.isDrawerOpen(GravityCompat.START)) {
      drawer.closeDrawer(GravityCompat.START);
    } else {
      super.onBackPressed();
    }
  }


  @SuppressWarnings("StatementWithEmptyBody")
  @Override
  public boolean onNavigationItemSelected(MenuItem item) {
    // Handle navigation view item clicks here.

    Fragment fragment = null;
    Class fragmentClass= null;

    switch (item.getItemId()) {
      case R.id.nav_ayat_fetcher:
        fragmentClass = AyahFetcherView.class;
        break;

      case R.id.nav_share:

        String message = "Download Muslim Companion Apps di [insert link google play]!\nBe a better muslim.";
        Intent share = new Intent(Intent.ACTION_SEND);
        share.setType("text/plain");
        share.putExtra(Intent.EXTRA_TEXT, message);
        Intent chooser = Intent.createChooser(share,"Share this apps");

        Log.i(getLocalClassName(), "onNavigationItemSelected: "+"Touchdown!");

        if(share.resolveActivity(getPackageManager()) != null){
          startActivity(chooser);
        }

        break;

      default:
        fragmentClass = AyahFetcherView.class;
    }

    if(fragmentClass != null){
      try {
        fragment = (Fragment) fragmentClass.newInstance();
      } catch (Exception e) {
        e.printStackTrace();
      }

      fragmentManager.beginTransaction().replace(R.id.view_fragment, fragment).commit();

      item.setChecked(true);
    }

    drawer.closeDrawer(GravityCompat.START);
    return true;
  }

}

File: app/src/main/java/io/github/hyuwah/muslimcompanionapp/presentation/View/MainActivity.kt
package io.github.hyuwah.muslimcompanionapp.presentation.View

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView
import io.github.hyuwah.muslimcompanionapp.R
import io.github.hyuwah.muslimcompanionapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        // Setup Toolbar & Drawer Layout
        setSupportActionBar(binding.appBarMain.toolbar)
        val toggle = ActionBarDrawerToggle(
                this, binding.drawerLayout, binding.appBarMain.toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        binding.drawerLayout.addDrawerListener(toggle)
        toggle.syncState()
        binding.navView.setNavigationItemSelectedListener(this)
        binding.navView.setCheckedItem(R.id.nav_ayat_fetcher)
        // Set default screen
        supportFragmentManager.beginTransaction().replace(R.id.view_fragment, AyahFetcherView()).commit()
    }

    override fun onBackPressed() {
        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
            binding.drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean { // Handle navigation view item clicks here.
        val fragment: Fragment?
        var fragmentClass: Class<*>? = null
        when (item.itemId) {
            R.id.nav_ayat_fetcher -> fragmentClass = AyahFetcherView::class.java
            R.id.nav_share -> {
                val message = "Download Muslim Companion Apps di [insert link google play]!\nBe a better muslim."
                val share = Intent(Intent.ACTION_SEND)
                share.type = "text/plain"
                share.putExtra(Intent.EXTRA_TEXT, message)
                val chooser = Intent.createChooser(share, "Share this apps")
                Log.i(localClassName, "onNavigationItemSelected: " + "Touchdown!")
                if (share.resolveActivity(packageManager) != null) {
                    startActivity(chooser)
                }
            }
            else -> fragmentClass = AyahFetcherView::class.java
        }
        if (fragmentClass != null) {
            try {
                fragment = fragmentClass.newInstance() as Fragment
                supportFragmentManager.beginTransaction().replace(R.id.view_fragment, fragment).commit()
                item.isChecked = true
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        binding.drawerLayout.closeDrawer(GravityCompat.START)
        return true
    }
}